src/middleware.rs

```diff
@@ -121,14 +120,13 @@ where
 
     fn call(&self, req: ServiceRequest) -> Self::Future {
         let service = Rc::clone(&self.service);
-        let extractor = Arc::clone(&self.extractor);
+        let extractor = Rc::clone(&self.extractor);
 
         Box::pin(async move {
             let permissions: Vec<String> = extractor.extract(&req).await?;
-            req.attach(permissions);
 
-            let fut = service.borrow_mut().call(req);
-            fut.await
+            req.extensions_mut().insert(AuthDetails::new(permissions));
```

Comment:
`attache` replaces this, allowing users to customize it directly to the request without going into the details of the data structure (`AuthDetails`).
For example when using another middleware, see: https://github.com/DDtKey/actix-web-grants/blob/bd934757fd6d75a8e448bc6733af418d1d70617b/examples/jwt-httpauth/src/main.rs#L43

Reply:
I reverted it back